runtime/src/main/scala/akka/grpc/internal/AkkaGrpcClient.scala

```diff
@@ -22,6 +24,10 @@ trait AkkaGrpcClient {
   def closed(): Future[Done]
 }
 
+trait AkkaGrpcClientFactory[T <: AkkaGrpcClient] {
```

Comment:
New class for creating `AkkaGrpcClient`s.

TODO: Add Scaladocs.

Reply:
I've been trying to simplify this a bit in https://github.com/akka/akka-grpc/commits/raboof-play-testing .

Reply:
The extensive use of `AkkaGrpcClient` in this PR seems to suggest it deserves to be outside of `internal` and in the `scaladsl`/`javadsl`, I'll do that in a separate PR